controllers/metallb_controller.go

```diff
@@ -160,91 +161,62 @@ func (r *MetalLBReconciler) SetupWithManager(mgr ctrl.Manager, bgpType string) e
 func (r *MetalLBReconciler) syncMetalLBResources(config *metallbv1beta1.MetalLB) error {
 	logger := r.Log.WithName("syncMetalLBResources")
 	logger.Info("Start")
-	data := render.MakeRenderData()
-
-	data.Data["SpeakerImage"] = os.Getenv("SPEAKER_IMAGE")
-	data.Data["ControllerImage"] = os.Getenv("CONTROLLER_IMAGE")
-	data.Data["FRRImage"] = os.Getenv("FRR_IMAGE")
-	data.Data["IsOpenShift"] = r.PlatformInfo.IsOpenShift()
-	data.Data["NameSpace"] = r.Namespace
-	data.Data["KubeRbacProxy"] = os.Getenv("KUBE_RBAC_PROXY_IMAGE")
-	data.Data["DeployKubeRbacProxies"] = os.Getenv("DEPLOY_KUBE_RBAC_PROXIES") == "true"
 
-	data.Data["LogLevel"] = metallbv1beta1.LogLevelInfo
-	if config.Spec.LogLevel != "" {
-		data.Data["LogLevel"] = config.Spec.LogLevel
-	}
-
-	var err error
-	data.Data["MetricsPort"], err = valueWithDefault("METRICS_PORT", 7472)
+	actionConfig := new(action.Configuration)
```

Comment:
I'd try to isolate this in a separate package
Ideally, a higher level function that takes the path of the charts and the metallb config and returns the list of objects, that can be unit tested in isolation.

Inside of it, I would split the values patching from the rest